lib/trackler/doc_file.rb

```diff
@@ -0,0 +1,88 @@
+class DocFile
+  DEFAULT_IMAGE_PATH = "/docs/img"
+
+  def self.find(basename:, track_dir:)
+    dir = File.join(track_dir, "docs")
+
+    [
+      MarkdownFile.new(basename: basename, docs_dir: dir),
+      OrgmodeFile.new(basename: basename, docs_dir: dir),
+    ].detect(&:exist?) || NullFile.new(basename)
+  end
+
+  attr_reader :basename, :dir
+  def initialize(basename:, docs_dir:)
+    @basename = basename
+    @dir = docs_dir
+  end
+
+  def render(image_path: DEFAULT_IMAGE_PATH)
+    body.gsub(img_src, img_dst(image_path))
+  end
+
+  def name
+    "%s.%s" % [basename, extension]
+  end
+
+  def extension
+    "md"
+  end
+
+  def exist?
+    File.exist?(path)
+  end
+
+  private
+
+  def body
+    File.read(path)
+  end
+
+  def path
+    File.join(dir, name)
+  end
+
+  def img_src
+    Regexp.new("]\\(%s" % DEFAULT_IMAGE_PATH)
+  end
+
+  def img_dst(image_path)
+    "](%s" % image_path.gsub(Regexp.new("/$"), "")
+  end
+end
+
+class OrgmodeFile < DocFile
+  def body
+    Orgmode::Parser.new(File.read(path)).to_markdown
+  end
+
+  def extension
+    "org"
+  end
+end
+
+class MarkdownFile < DocFile
+end
+
+class NullFile
```

Comment:
> In Ruby I thought the type is irrelevant, as long as it responds to the expected messages.

Largely, It's still helpful to provide this information for debugging purposes.
It's nice to be able to call `.class` on an object and get an idea of what it is supposed to do/be.

> The private methods. It feels worse to me to have a bunch of behavior and then override it to be nothing, than to start from scratch and add empty behavior.

I feel the opposite. If one finds that they need to override a lot of behavior this 
is smell that there is something that could be improved in the underlying abstraction.